features/android/support/app_life_cycle_hooks.rb

```diff
@@ -1,26 +1,25 @@
+require 'calabash-android/management/adb'
 require 'calabash-android/operations'
 
 
-INSTALLATION_STATE = {
-    :installed => false
-}
-
 Before do |scenario|
   scenario_tags = scenario.source_tag_names
-  if !INSTALLATION_STATE[:installed]
-      uninstall_apps
-      install_app(ENV['TEST_APP_PATH'])
-      install_app(ENV['APP_PATH'])
-      INSTALLATION_STATE[:installed] = true
-  end
 
-  if scenario_tags.include?('@reinstall')
-    clear_app_data
+  # for Amazon Device Farm no need to reinstall
+  # it gives a fresh device every time for each .feature file
+  # if you have more scenario in a .feature, you may need this
+  if scenario_tags.include?('@reinstall') and ENV["LOCALRUN"] != "true"
```

Comment:
The convention in Calabash is check against `"1"` or `"0"` when testing boolean environment variables.

> and vs &&

Avoid surprises and always use `&&`.

```suggestion
  if scenario_tags.include?('@reinstall') && ENV["LOCALRUN"] != "1"
```